### PR TITLE
executor: refine the check of onlyFullGroupBy when groupByItem is parenthesesExpr or UnaryPlusExpr (#13655)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
cherry-pick #13655 

### What problem does this PR solve? 
fix issue #13652  

### What is changed and how it works?
Add the check for ParenthesesExpr and UnaryPlusExpr in checkOnlyFullGroupByWithGroupClause

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

Release note

Make the check of SQL Mode only_full_group_by correct when the group by items is wrapped with parentheses.

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -1734,14 +1734,15 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
 	gbyExprs := make([]ast.ExprNode, 0, len(sel.Fields.Fields))
 	schema := p.Schema()
 	for _, byItem := range sel.GroupBy.Items {
+		expr := getInnerFromParenthesesAndUnaryPlus(byItem.Expr)
 		if colExpr, ok := byItem.Expr.(*ast.ColumnNameExpr); ok {
```
Comment (lamxTyler) [inline planner/core/logical_plan_builder.go]: It should be `expr`?

Comment (zz-jason) [reply planner/core/logical_plan_builder.go]: @XuHuaiyu PTAL this comment

**Pull request merged**